src/history/history.go

```diff
@@ -94,18 +94,21 @@ func StoreHealth(hc *HealthCheck) {
 	go writeHealth(hc)
 }
 
-// GetState returns the current state of the service
-func GetState(repoURL string) State {
+// GetState returns the current state of the service and a tag identifying it
+func GetState(repoURL string) (State, string) {
 	mux.Lock()
 	defer mux.Unlock()
 	if _, ok := history[repoURL]; !ok {
 		history[repoURL] = NewService()
 	}
-	return *history[repoURL].Current
+	tag := makeTag(repoURL, *history[repoURL].Current)
```

Comment:
Why do you need to make a tag everytime? Just make it at the SetState function, store it and return it.

Reply:
If you store it in the history map, You will also need to check the validity of the tag when we init the map in https://github.com/devclub-iitd/DeployBot/blob/master/src/history/history.go#L30

Reply:
What should be done if the tag isn't valid? Since it shouldn't happen in the normal usage, I was thinking of just throwing an error, but that won't play well with the existing history file. Maybe have it set the tag when there is no tag field in the JSON file entry?

Reply:
Why will it not work with the existing history file? You can check for tag presence in the init function in https://github.com/devclub-iitd/DeployBot/blob/master/src/history/history.go#L17
and if it is not present there, just generate the tag again. Don't check for tag validity in this function. This function is atomic due to a lock, so it should have as few instructions as possible.